convert2rhel/utils.py

```diff
@@ -125,6 +125,123 @@ def restart_system():
         loggerinst.warning("In order to boot the RHEL kernel, restart of the system is needed.")
 
 
+def _normalize_subprocess_output(output, skip_stdout_decode):
+    """Internal function to normalize the output from subprocess.
+
+    .. note::
+        This function will try to handle the case where the output is bytes
+        and not str, but, if the caller requested that the output should not
+        be decoded, then the function will return the data in bytes and not
+        an str.
+
+    :param output: The output of a subprocess call.
+    :type output: bytes
+    :param skip_stdout_decode: Flag to indicate that the output will not be
+        decoded as utf-8. Useful if the function is working with binary outputs.
+    :type skip_stdout_decode: bool
+    :return: A normalized output, either in an string format or in bytes format.
+    :rtype: str | bytes
+    """
+    normalized_output = ""
+    for line in iter(output, b""):
+        if skip_stdout_decode:
+            if not isinstance(normalized_output, bytes):
+                normalized_output = normalized_output.encode()
+
+            normalized_output += line
+        else:
+            normalized_output += line.decode("utf-8")
+
+    return normalized_output
+
+
+def run_piped_subprocess(
```

Comment:
@abadger, not sure if that's the correct and best way of coming up with this function, but please, if you review it, let me know what I could improve here.